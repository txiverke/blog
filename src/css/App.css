/** GLOBAL VARIABLES **/
@import './_variables.css';

/** LAYOUT STYLES **/
@import './layout/Body.css';
@import './layout/Content.css';
@import './layout/Footer.css';
@import './layout/Header.css';

/** COMPONENT STYLES **/
@import './component/Article.css';
@import './component/Background.css';
@import './component/Button.css';
@import './component/Form.css';
@import './component/List.css';
@import './component/Loader.css';
@import './component/Search.css';

/** PROPERTY STYLES **/
@import './property/BoxModel.css';
@import './property/Grid.css';
@import './property/Image.css';
@import './property/Text.css';
