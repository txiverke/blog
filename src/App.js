// @flow

import React from 'react';
import Helmet from 'react-helmet'

import Header from './components/Header'
import Routes from './Routes'
import Footer from './components/Footer'
import { getLanguage } from './utils/helpers'
import { getDictionary } from './utils/dictionary'

import './css/App.css'

const currentYear = new Date()
const DIC = getDictionary(getLanguage())

const App = () => {
  return (
    <main className="app-root">
      <Helmet titleTemplate={`%s | Xavier Vilà - front-end developer`} defaultTitle="Xavi Vilà - front-end developer" >
        <link rel="canonical" href="http://xaviervila.tech" />
        <meta name="description" content={DIC.DESCRIPTION}  />
      </Helmet>
      <Header />
      <Routes />
      <Footer year={currentYear} />
    </main>
  )
}
export default App
