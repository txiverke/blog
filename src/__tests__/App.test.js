// @flow

import React from 'react'
import { shallow } from 'enzyme'
import App from '../App'

describe('<App />', () => {
  it('should render itself', () => {
    const component = shallow(<App />)
    expect(component).toMatchSnapshot()
  })
})