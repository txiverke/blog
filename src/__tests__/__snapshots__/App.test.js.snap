// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render itself 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HelmetWrapper
          defaultTitle="Xavi Vilà - front-end developer"
          defer={true}
          encodeSpecialCharacters={true}
          titleTemplate="%s | Xavier Vilà - front-end developer"
>
          <link
                    href="http://xaviervila.tech"
                    rel="canonical"
          />
          <meta
                    content="Hi I'm Xavi, a front-end developer from Lleida (Catalunya) currently living with my wife and our dog and two cats in Ferndale (Michigan)."
                    name="description"
          />
</HelmetWrapper>,
        <Connect(Component) />,
        <Routes />,
        <Footer />,
      ],
      "className": "app-root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <link
              href="http://xaviervila.tech"
              rel="canonical"
/>,
            <meta
              content="Hi I'm Xavi, a front-end developer from Lleida (Catalunya) currently living with my wife and our dog and two cats in Ferndale (Michigan)."
              name="description"
/>,
          ],
          "defaultTitle": "Xavi Vilà - front-end developer",
          "defer": true,
          "encodeSpecialCharacters": true,
          "titleTemplate": "%s | Xavier Vilà - front-end developer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "href": "http://xaviervila.tech",
              "rel": "canonical",
            },
            "ref": null,
            "rendered": null,
            "type": "link",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "content": "Hi I'm Xavi, a front-end developer from Lleida (Catalunya) currently living with my wife and our dog and two cats in Ferndale (Michigan).",
              "name": "description",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HelmetWrapper
            defaultTitle="Xavi Vilà - front-end developer"
            defer={true}
            encodeSpecialCharacters={true}
            titleTemplate="%s | Xavier Vilà - front-end developer"
>
            <link
                        href="http://xaviervila.tech"
                        rel="canonical"
            />
            <meta
                        content="Hi I'm Xavi, a front-end developer from Lleida (Catalunya) currently living with my wife and our dog and two cats in Ferndale (Michigan)."
                        name="description"
            />
</HelmetWrapper>,
          <Connect(Component) />,
          <Routes />,
          <Footer />,
        ],
        "className": "app-root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <link
                href="http://xaviervila.tech"
                rel="canonical"
/>,
              <meta
                content="Hi I'm Xavi, a front-end developer from Lleida (Catalunya) currently living with my wife and our dog and two cats in Ferndale (Michigan)."
                name="description"
/>,
            ],
            "defaultTitle": "Xavi Vilà - front-end developer",
            "defer": true,
            "encodeSpecialCharacters": true,
            "titleTemplate": "%s | Xavier Vilà - front-end developer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "href": "http://xaviervila.tech",
                "rel": "canonical",
              },
              "ref": null,
              "rendered": null,
              "type": "link",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "content": "Hi I'm Xavi, a front-end developer from Lleida (Catalunya) currently living with my wife and our dog and two cats in Ferndale (Michigan).",
                "name": "description",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
